packages/superset-ui-core/src/query/extractQueryFields.ts

```diff
@@ -46,26 +45,46 @@ export default function extractQueryFields(
   };
   const finalQueryFields: QueryFields = {
     columns: [],
-    groupby: [],
     metrics: [],
   };
+  const { query_mode: queryMode, include_time: includeTime, ...restFormData } = formData;
 
-  Object.entries(formData).forEach(([key, value]) => {
-    const normalizedKey = queryFieldAliases[key] || key;
-    if (normalizedKey in finalQueryFields) {
-      if (normalizedKey === 'metrics') {
-        finalQueryFields[normalizedKey] = finalQueryFields[normalizedKey].concat(value);
-      } else {
-        // currently the groupby and columns field only accept pre-defined columns (string shortcut)
-        finalQueryFields[normalizedKey] = finalQueryFields[normalizedKey]
-          .concat(value)
-          .filter(x => typeof x === 'string' && x);
-      }
+  Object.entries(restFormData).forEach(([key, value]) => {
+    // ignore `null` or `undefined` value
+    if (value == null) {
+      return;
+    }
+
+    let normalizedKey: string = queryFieldAliases[key] || key;
+
+    // ignore groupby and metrics when in raw records mode
+    if (
+      queryMode === QueryMode.raw &&
+      (normalizedKey === 'groupby' || normalizedKey === 'metrics')
+    ) {
+      return;
+    }
+    // ignore columns when (specifically) in aggregate mode.
+    if (queryMode === QueryMode.aggregate && normalizedKey === 'columns') {
+      return;
+    }
+    // groupby has been deprecated: https://github.com/apache/superset/pull/9366
+    if (normalizedKey === 'groupby') {
+      normalizedKey = 'columns';
+    }
+    if (normalizedKey === 'metrics') {
```

Comment:
I try to avoid `switch` nowadays since Python has taught me [`switch` is just a syntactic sugar for if else's](https://stackoverflow.com/questions/46701063/why-doesnt-python-have-switch-case). Plus I hate it when I forgot to call `break`.